doc/usecases/core/UC_key_clear.md

```diff
@@ -0,0 +1,24 @@
+# Use Case: `Key` clear
+
+## Summary
+
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Clearing contents of a `Key`
+- **Status:** Implemented
+
+## Scenarios
+
+- **Precondition:**
+  - [`Key` has been created](UC_key_create.md)
+- **Main success scenario:**
+  - Caller requests the fields of a `Key` should get cleard
```

Comment:
```suggestion
  - Caller requests the data of a `Key` should get cleared
```

Data as in "what's publicly visible", some internal fields could remain or be set to specific values (e.g. sync flag).